### ISSUE TITLE
SortingAlgorithm carries across separate grids

### ISSUE THREAD

**Issue by pitops** [angular-ui/ui-grid#6406]
I have noticed an issue when using sortingAlgorithm on a field that exists in two grids.

Say the field is called 'power' in both of these grids and they both have different sortingAlgorithm, the sortingAlgorithm used in both cases is the one used in the first instantiated grid.

I found another person on stackoverflow who had the same issue, solved it by changing the name of the field to be different in the two grids but this is not an option for my case. Is there a workaround?

```
{
        field: 'power',
        type: 'number',
        enableFiltering: false,
        headerCellClass: 'n-power-header',
       ....
        sortingAlgorithm: (a, b, rowA, rowB, direction) => {
          ..sort code
        }
}
```

**Comment by pitops** [comment]
We found a solution that is cleaner than changing the field name and seems to work consistently for anyone who is having the same issue. 

first decide which of the instantiated grid(s) you wish to change. If you have two instantiated grids you only need to change one else you need to only leave one unchanged.

### Change the field property to hold a function name
```
{
        field: 'myFunction()',
        type: 'number',
        enableFiltering: false,
        headerCellClass: 'n-power-header',
       ....
        sortingAlgorithm: (a, b, rowA, rowB, direction) => {
          ..sort code
        }
}
```

In your controller where the grid is instantiated - after instantiation write a function like the following

```
angular.forEach($scope.GridData, (row) => {
        if (!row.myFunction) {
          row.myFunction = () => row.power
        }
      })
```

Basically all the function does is return the field and this way it works around the issue. If you have data that come in periodically i.e websockets remember to attach this function to the update event as well.

Hope that helps

**Comment by zamb3zi** [comment]
Not sure why this is closed. It's still a bug with no ideal workaround.

I found I had to change the field names as @pitops workaround didn't work for me.
